code/RedirectedURLHandler.php

```diff
@@ -20,83 +43,92 @@ class RedirectedURLHandler extends Extension {
 	 */
 	protected function arrayToLowercase($vars) {
 		$result = array();
-
 		foreach($vars as $k => $v) {
 			if(is_array($v)) {
 				$result[strtolower($k)] = $this->arrayToLowercase($v);
 			} else {
-			    $result[strtolower($k)] = strtolower($v);
-            }
+				$result[strtolower($k)] = strtolower($v);
+			}
 		}
-
 		return $result;
 	}
 
 	/**
 	 * @throws SS_HTTPResponse_Exception
 	 */
 	public function onBeforeHTTPError404($request) {
-		$base = strtolower($request->getURL());
+		$caseInsensitiveMatching = $this->config()->get('case_insensitive_matching');
```

Comment:
I think `$this->config()` is a SilverStripe 4 thing - you should probably use `Config::inst()->get(get_class($this), 'case_insensitive_matching')` for SS3

Reply:
Looking at the API docs for both SS3 and SS4 and `Extension` doesn't contain a `config()` method and doesn't extend any other classes, whereas everything else usually extends all the way to `Object` which does have the method. I can't seem to find any docs or issues regarding this so I'll switch back to using `Config::inst()->get(get_class($this), 'case_insensitive_matching')`

Reply:
Ah, sorry for saying to do it the other way. if it's an extension is should be pulling it off of the extended object (ie: off of `$this->owner`)

so it *really* should be `$this->owner->config()->get(...)`

Reply:
OK sweet, then if we're getting the owner config, does that mean the option will need to be set per-class that's using `RedirectedURLHandler` as an extension, rather than a single global option `RedirectedURLHandler`? e.g
```
Controller::add_extension('RedirectedURLHandler');
Controller::config()->case_insensitive_matching = true;
```

Reply:
Instances with the extension applied will inherit the config from the extension